packages/backend-common/src/errors.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * A set of common business logic errors.
+ *
+ * The error handler middleware understands these and will translate them to
+ * well formed HTTP responses.
+ *
+ * While these are intentionally analogous to HTTP errors, they are not
+ * intended to be thrown by the request handling layer. In those places, please
+ * use e.g. the http-errors library.
+ */
+
+class CustomErrorBase extends Error {
+  readonly cause?: Error;
+
+  constructor(constructor: Function, message?: string, cause?: Error) {
+    super(message);
+    Object.setPrototypeOf(this, constructor.prototype);
+    Error.captureStackTrace(this, constructor);
+    this.name = this.constructor.name;
+    this.cause = cause;
+  }
+
+  toString() {
+    let result = super.toString();
+
+    if (this.cause) {
+      result += `; caused by ${this.cause.toString()}`;
+    }
+
+    return result;
+  }
+}
+
+/**
+ * The request is malformed and cannot be processed.
+ */
+export class BadRequestError extends CustomErrorBase {
```

Comment:
Not sure if we should have this tbh, might be leaking a bit too much http, but fine with it for now at least.

Reply:
Yeah i just want SOME form of path for reporting that the input was invalid (something that ultimately becomes a 400).

Reply:
Yup, thinking about alternatives like `InputError` or `BadValueError`

Reply:
Wanna go for `InputError`? I'm game!

Reply:
![image](https://user-images.githubusercontent.com/4984472/80978142-f3984c00-8e25-11ea-9d35-b82327973ea2.png)

Reply:
done